### PR TITLE
[SystemUiController] Add API 30 workaround for system bar appearance

### LINKED ISSUE #881: SystemUiController.setStatusbarColor not changed icons color in status bar
## Describe the bug
SystemUiController.setStatusbarColor not changed icons color in status bar if android:windowLightStatusBar is "true" in themes.xml. If  android:windowLightStatusBar is "false" SystemUiController.setStatusbarColor works correct. Is it correct behavior?
## To Reproduce
1. Go to themes.xml and add `<item name="android:windowLightStatusBar">true</item>` to current theme which selected in AndroidManifiest.xml
2. Try to change icons color in status bar to light:
```
SideEffect {
        systemUiController.setStatusBarColor(
            color = statusBarColor,
            darkIcons = false
        )
    }
```
3. Icons in status bar still dark.
## Expected behavior
Icons color should be light

## Screenshots?
examplecolor

## Environment:
 - Android OS version: Android 11
 - Device: OnePlus 9R, OnePlus 3
 - Accompanist version: v0.21.2-beta

### LINKED ISSUE #918: [System UI Controller] Preview is not rendering
Preview started throwing an exception after commit bb866bf3a1e96eb14a746649dcc9bb39878542af

```
java.lang.IllegalArgumentException: The Compose View must be hosted in an Activity with a Window!  at
com.google.accompanist.systemuicontroller.AndroidSystemUiController.<init>(SystemUiController.kt:177)   at
com.google.accompanist.systemuicontroller.SystemUiControllerKt.rememberSystemUiController(SystemUiController.kt:165)
```

https://github.com/google/accompanist/blob/ce11b260f2d2e5d69d515833b4437cdb2c67e82d/systemuicontroller/src/main/java/com/google/accompanist/systemuicontroller/SystemUiController.kt#L177

### PR DESCRIPTION
**Pull request by alexvanyo**
This should actually fix #881 fully and #949, for the case where the theme has `android:windowLightStatusBar=true` and trying to set it to `false` (via `statusBarDarkContentEnabled = false` or `darkIcons = false`).

There was also a somewhat similar issue for the navigation bars, where the scrim wouldn't be updated while toggling the navigation bar dark icons on API 30, which I believe is https://issuetracker.google.com/issues/215832843 but hasn't been reported here yet as an issue.

The fix for both of these seems to be setting the corresponding `systemUiVisibility` flags anyway, in addition to calling the normal flags. I'm planning to upstream these changes as well, so that we don't have to use these workarounds with a future version of `androidx.core`.

This fix also made the `systemUiController` optional again, so that it will work when using it in a `@Preview` (and fixes #918)

BEFORE:

https://user-images.githubusercontent.com/4217560/151279075-a90b5dc7-0d8d-44fc-ab5a-47401fca5756.mp4

AFTER:

https://user-images.githubusercontent.com/4217560/151279079-54da2140-643a-45ba-a311-736d44a1a777.mp4

### CODE REVIEW

**Comment by hrach** [comment]
Yeah, I was studying if the navigation bar issue is accompanist's, but ended up creating the issue for androidx.core :) 

Thanks for workarounding it here!

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -54,7 +54,7 @@ coil-gif = { module = "io.coil-kt:coil-gif", version.ref = "coil" }
 coil-compose = { module = "io.coil-kt:coil-compose", version.ref = "coil" }
 
 androidx-appcompat = "androidx.appcompat:appcompat:1.4.1"
-androidx-core = "androidx.core:core-ktx:1.7.0"
+androidx-core = "androidx.core:core-ktx:1.8.0-alpha03"
```
Comment (bentrengrove) [inline gradle/libs.versions.toml:57]: What change did you need to pull in the alpha version?

**Review by bentrengrove — DISMISSED** [review]
LGTM. Just checking we actually need core-ktx alpha. Definitely agree we need to try and upstream this rather than having a workaround here.

File: gradle/libs.versions.toml (hunk 1 above)
Comment (alexvanyo) [reply gradle/libs.versions.toml]: Ah, this wasn't strictly necessary, but `ViewCompat.getWindowInsetsController` will be the single non-deprecated way to get the `WindowInsetsControllerCompat` in the upcoming release

**Comment by alexvanyo** [comment]
@bentrengrove I just merged in the upstream fix in core-ktx, so it will be available in a snapshot build soon and probably in the next alpha: https://android-review.googlesource.com/c/platform/frameworks/support/+/1962461

Do we want to merge in the workaround here anyway? (applying the workaround twice should be harmless)
Or wait for it to be released in `core`?
And if it is released in `core`, do we want to transitively bump it via `systemuicontroller` (even if it is still `1.8.0-alphaXX`.

Part of this PR should still be merged at some point, both to fix `@Preview`s and to update to the non-deprecated `WindowInsetsControllerCompat` creation method.

**Comment by bentrengrove** [comment]
Nice work upstreaming! In that case I think we wait for it to be released in core and then point to that version once it's out in alpha. I don't think we should be pointing to snapshot releases.

Do you want to close this one and open a new PR just with the partial changes you mentioned still needing to go in?

**Comment by alexvanyo** [comment]
Agreed, was able to verify that the snapshot will fix it with the latest commit. Will close and open up a new PR.

**Comment by prime-zs** [comment]
I am using accompinist version 0.26.2-beta, recently the ui is not updating with icon color change. I am testing it in android 9

**Comment by alexvanyo** [comment]
@prime-zs please create a new bug if you're running into an issue with the latest version!

**Pull request closed without merging**